musig/src/errors.rs

```diff
@@ -0,0 +1,22 @@
+/// Represents an error in key aggregation, signing, or verification.
+#[derive(Fail, Clone, Debug, Eq, PartialEq)]
+pub enum MusigError {
+    /// This error occurs when a point is not a valid compressed Ristretto point
+    #[fail(display = "Point decoding failed.")]
+    InvalidPoint,
+
+    /// This error occurs when a signature share fails to verify
+    #[fail(display = "Share #{:?} failed to verify correctly", pubkey)]
+    ShareError {
+        /// The pubkey corresponding to the share that failed fo verify correctly
+        pubkey: [u8; 32],
+    },
+
+    /// This error occurs when an individual point operation failed.
+    #[fail(display = "Point operation failed.")]
+    PointOperationFailed,
+
+    /// This error occurs when a function is called with bad arguments.
+    #[fail(display = "Bad arguments")]
```

Comment:
Is there style guidance about these strings? Should they be capitalized? Some of these end in periods and some don't, should they be consistent?

Reply:
I couldn't find an official style guide, but just looking through the examples, it seems like capitalize + no period at the end is the norm. I'll update that!
Example: https://doc.rust-lang.org/std/fmt/struct.Error.html